### ISSUE TITLE
Supervisor not recreating containers after containers prune

### ISSUE THREAD

**Issue by unlucio** [home-assistant/hassio#1095]
**Home Assistant release with the issue:**
homeassistant/armhf-hassio-supervisor      164
omeassistant/raspberrypi3-homeassistant   0.92.1

Docker version 18.09.0, build 4d60db4

**Operating environment (HassOS/Generic):**
```sh
root@raspberrypi:/mnt/bigstore/hass # uname -a
Linux raspberrypi 4.14.98-v7+ #1200 SMP Tue Feb 12 20:27:48 GMT 2019 armv7l GNU/Linux
```

**Supervisor logs:**
```sh
19-05-20 21:00:43 INFO (MainThread) [__main__] Initialize Hass.io setup
19-05-20 21:00:43 INFO (MainThread) [__main__] Setup HassIO
19-05-20 21:00:43 INFO (SyncWorker_0) [hassio.docker.supervisor] Attach to Supervisor homeassistant/armhf-hassio-supervisor with version 164
19-05-20 21:00:43 INFO (SyncWorker_0) [hassio.docker.supervisor] Connect Supervisor to Hass.io Network
19-05-20 21:00:45 INFO (MainThread) [hassio.utils.gdbus] Introspect org.freedesktop.systemd1 on /org/freedesktop/systemd1
19-05-20 21:00:45 INFO (MainThread) [hassio.utils.gdbus] Connect to dbus: org.freedesktop.systemd1 - /org/freedesktop/systemd1
19-05-20 21:00:45 INFO (MainThread) [hassio.utils.gdbus] Introspect org.freedesktop.hostname1 on /org/freedesktop/hostname1
19-05-20 21:00:45 INFO (MainThread) [hassio.utils.gdbus] Connect to dbus: org.freedesktop.hostname1 - /org/freedesktop/hostname1
19-05-20 21:00:45 INFO (MainThread) [hassio.utils.gdbus] Introspect de.pengutronix.rauc on /
19-05-20 21:00:45 ERROR (MainThread) [hassio.utils.gdbus] DBus return error: b'Error: GDBus.Error:org.freedesktop.DBus.Error.ServiceUnknown: The name de.pengutronix.rauc was not provided by any .service files\n'
19-05-20 21:00:45 WARNING (MainThread) [hassio.dbus.rauc] Can't connect to rauc
19-05-20 21:00:45 INFO (MainThread) [hassio.host.info] Update local host information
19-05-20 21:00:45 INFO (MainThread) [hassio.utils.gdbus] Call org.freedesktop.DBus.Properties.GetAll on /org/freedesktop/hostname1
19-05-20 21:00:45 INFO (MainThread) [hassio.host.services] Update service information
19-05-20 21:00:45 INFO (MainThread) [hassio.utils.gdbus] Call org.freedesktop.systemd1.Manager.ListUnits on /org/freedesktop/systemd1
19-05-20 21:00:45 INFO (MainThread) [hassio.host.apparmor] Load AppArmor Profiles: {'hassio-supervisor', 'core_dnsmasq'}
19-05-20 21:00:45 INFO (MainThread) [hassio.host.services] Reload local service hassio-apparmor.service
19-05-20 21:00:45 INFO (MainThread) [hassio.utils.gdbus] Call org.freedesktop.systemd1.Manager.ReloadOrRestartUnit on /org/freedesktop/systemd1
19-05-20 21:00:46 INFO (SyncWorker_2) [hassio.docker.interface] Attach to homeassistant/raspberrypi3-homeassistant with version 0.92.1
19-05-20 21:00:46 INFO (MainThread) [hassio.store.git] Load add-on /data/addons/git/a0d7b954 repository
19-05-20 21:00:46 INFO (MainThread) [hassio.store.git] Load add-on /data/addons/core repository
19-05-20 21:00:46 INFO (MainThread) [hassio.store] Load add-ons from store: 59 all - 59 new - 0 remove
19-05-20 21:00:46 INFO (MainThread) [hassio.addons] Found 6 installed add-ons
19-05-20 21:00:46 INFO (SyncWorker_1) [hassio.docker.interface] Attach to hassioaddons/grafana-armv7 with version v2.1.1
19-05-20 21:00:46 INFO (SyncWorker_3) [hassio.docker.interface] Attach to hassioaddons/portainer-armv7 with version v0.6.1
19-05-20 21:00:46 INFO (SyncWorker_2) [hassio.docker.interface] Attach to homeassistant/armv7-addon-configurator with version 3.3
19-05-20 21:00:46 INFO (SyncWorker_5) [hassio.docker.interface] Attach to hassioaddons/node-red-armv7 with version v3.1.1
19-05-20 21:00:46 INFO (SyncWorker_6) [hassio.docker.interface] Attach to homeassistant/armv7-addon-dnsmasq with version 1.2
19-05-20 21:00:46 INFO (SyncWorker_7) [hassio.docker.interface] Attach to hassioaddons/influxdb-armv7 with version v3.0.5
19-05-20 21:00:47 INFO (MainThread) [hassio.updater] Fetch update data from https://s3.amazonaws.com/hassio-version/stable.json
19-05-20 21:00:47 INFO (MainThread) [hassio.snapshots] Found 0 snapshot files
19-05-20 21:00:47 INFO (MainThread) [hassio.discovery] Load 0 messages
19-05-20 21:00:47 INFO (MainThread) [hassio.ingress] Load 0 ingress session
19-05-20 21:00:47 INFO (MainThread) [__main__] Run Hass.io
19-05-20 21:00:47 INFO (MainThread) [hassio.misc.dns] Start DNS port forwarding for host add-ons
19-05-20 21:00:47 INFO (MainThread) [hassio.api] Start API on 172.30.32.2
19-05-20 21:00:47 INFO (MainThread) [hassio.addons] Phase 'initialize' start 0 add-ons
19-05-20 21:00:47 INFO (MainThread) [hassio.core] Hass.io reboot detected
19-05-20 21:00:47 INFO (MainThread) [hassio.tasks] All core tasks are scheduled
19-05-20 21:00:47 INFO (MainThread) [hassio.core] Hass.io is up and running
```

**Description of problem:**
I had to prune my containers (disk space):
- stopped hassio-service
- docker containers prune
- started hassio-services
now I've been waiting for a good half an hour but no other container was created other than `hassio_supervisor`.
I tried running the `https://github.com/home-assistant/hassio-installer` script again but nothing changed: still a lonely container.
Surprisingly enough the logs say the supervisor is attaching to my addons,
ie `[hassio.docker.interface] Attach to homeassistant/armv7-addon-configurator with version 3.3`
but this is my current `docker ps -a`
```sh
# docker ps -a
CONTAINER ID        IMAGE                                       COMMAND                  CREATED             STATUS                        PORTS                    NAMES
3e678c51a781        homeassistant/armhf-hassio-supervisor       "python3 -m hassio"      12 minutes ago      Up 12 minutes                                          hassio_supervisor
9df6d4a851e7        matrixanger/rpi-ikev2                       "/bin/sh -c /usr/bin…"   7 weeks ago         Exited (137) 6 weeks ago                               ikev2
2f9f25a808b8        dnsupdater                                  "/usr/local/bin/npm …"   3 months ago        Exited (0) 3 minutes ago                               updater
```
so I'm not sure who's that guy talking to 🤔

**Comment by pvizeli** [comment]
Well, we don't support manual interaction with docker on our containers.

Rerun the installer should fix your problem.

**Comment by unlucio** [comment]
> Rerun the installer should fix your problem.
> > I tried running the https://github.com/home-assistant/hassio-installer script again but nothing changed: still a lonely container.

I guessed for some reason only the installer is able to create the `home-assistant` container, so running the installer is among the things I tried but it didn't help :/
Last time this happened I had to hard reset my docker environment (more or less `rm -fr /var/lib/docker/*`) in order for the installer to actually fully work. his was months ago, and since I noticed a number of updates since then, I thought (what in my eyes looks like a problem) might have been resolved.


> Well, we don't support manual interaction with docker on our containers.

🤔from my (the user) perspective: the pi is mine, the docker daemon is mine, hass.io crated containers are among my containers (not alone) and I shall be allowed to do what I need (and docker allows) to do with my containers. As a user, if differently warned, I should be able to do what's usual and standard, and deleting containers is both pretty usual and standard. 
I understand that laying out the recovery logic might be tedious, out of scope or differently prioritized on the roadmap, but perhaps for the time being it might be the case to loudly warn the user of this (non-trivial) requirement.

I'm not a python ninja by any mean (I use it when I have, at work, but I prefer other languages) but if you roughly point me in the right direction where in the source tree `home-assistant`'s container and other dependencies are attached I'm keen perhaps to send a PR on this when I find some time.
